### PR TITLE
cli test for security certificates

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

This include cli tests for:

- importing security certificates

- listing certificates
- removing certificates

## Related Issue

- Fixes <issue_link>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/cliMain/securityCertificates.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
+@cli
+Feature: security certificates
+  As an admin
+  I want to be able to manage the ownCloud security certificates
+  So that I can ensure the proper encrpytion mechanism
+
+  Scenario: List security certificates when no certificates are installed
+    When the administrator invokes occ command "security:certificates"
+    Then the command should have been successful
+    And the command output table should contain the following text:
```
Comment (dpakach) [inline tests/acceptance/features/cliMain/securityCertificates.feature]: I think we need another step here
```
Then the command output table should not contain any values
```

File: tests/acceptance/features/cliMain/securityCertificates.feature [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      | table_column |
+    
+  Scenario: List security certificates when certificates are installed
+    When the administrator invokes occ command "security:certificates:import tests/data/certificates/goodCertificate.crt "
```
Comment (dpakach) [inline tests/acceptance/features/cliMain/securityCertificates.feature]: Maybe use Given steps here.

File: tests/acceptance/features/cliMain/securityCertificates.feature [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    Then the command should have been successful
+    When the administrator invokes occ command "security:certificates"
+    Then the command should have been successful
+    And the command output table should contain the following text:
+    | table_column         |
+    | goodCertificate.crt  |
+    | badCertificate.crt   |
+
+  Scenario: Remove a security certificate
+    When the administrator invokes occ command "security:certificates:import tests/data/certificates/goodCertificate.crt "
```
Comment (dpakach) [inline tests/acceptance/features/cliMain/securityCertificates.feature]: Also given here

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 4]
```diff
@@ -55,6 +62,20 @@ public function invokingTheCommand($cmd) {
...
+	/**
+	 * @Given the administrator imports security certificate form the path :arg1
+	 *
+	 * @param string $path
+	 *
+	 * @return void
+	 */
+	public function theAdministratorImportsSecurityCertificateFormThePath($path) {
+		$this->invokingTheCommand("security:certificates:import " . $path);
+		$componenets = \explode("/", $path);
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/OccContext.php]: is this supposed to be `components`?

File: tests/acceptance/features/cliMain/securityCertificates.feature [hunk 5]
```diff
@@ -0,0 +1,35 @@
+@cli
+Feature: security certificates
+  As an admin
+  I want to be able to manage the ownCloud security certificates
+  So that I can ensure the proper encrpytion mechanism
+
+  Scenario: List security certificates when certificates are installed
+    Given the administrator imports security certificate form the path "tests/data/certificates/goodCertificate.crt"
```
Comment (dpakach) [inline tests/acceptance/features/cliMain/securityCertificates.feature]:
```suggestion
    Given the administrator has imported security certificate form the path "tests/data/certificates/goodCertificate.crt"
```

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 6]
```diff
@@ -845,6 +866,19 @@ public function theSystemConfigKeyFromLastCommandOutputShouldContainValue(
...
+	 * This will run after EVERY scenario.
+	 * It will set the properties for this object.
+	 *
+	 * @AfterScenario
+	 *
+	 * @return void
+	 */
+	public function removeImportedCertificates() {
+		foreach ($this->importedCertificates as $certificate) {
+			$this->invokingTheCommand("security:certificates:remove " . $certificate);
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/OccContext.php]: Should we also check if the command is successful?

File: tests/acceptance/features/bootstrap/OccContext.php [hunk 7]
```diff
@@ -55,6 +61,21 @@ public function invokingTheCommand($cmd) {
...
+	 * @When the administrator imports security certificate form the path :path
+	 * @Given the administrator has imported security certificate form the path :path
+	 *
+	 * @param string $path
+	 *
+	 * @return void
+	 */
+	public function theAdministratorImportsSecurityCertificateFormThePath($path) {
+		$this->invokingTheCommand("security:certificates:import " . $path);
+		$pathComponenets = \explode("/", $path);
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/OccContext.php]:
```suggestion
		$pathComponents = \explode("/", $path);
```

**Comment by phil-davis** [comment]
@HariBhandari07 drone is still running the smb_windows job that does not work: https://drone.owncloud.com/owncloud/core/18037/251
You need to rebase to the really latest core master.

**Review by individual-it — APPROVED** [review]

**Comment by HariBhandari07** [comment]
Backport on #35432

**Comment by phil-davis** [comment]
Backport `stable10` #35435

**Pull request merged**